src/recommendpolicy/helperFunctions.go

```diff
@@ -7,40 +7,110 @@ import (
 	"strings"
 
 	"github.com/accuknox/auto-policy-discovery/src/admissioncontrollerpolicy"
-
+	"github.com/accuknox/auto-policy-discovery/src/cluster"
+	cfg "github.com/accuknox/auto-policy-discovery/src/config"
+	"github.com/accuknox/auto-policy-discovery/src/types"
 	"github.com/clarketm/json"
 	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
 	v1 "k8s.io/api/apps/v1"
 	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/watch"
-
-	"github.com/accuknox/auto-policy-discovery/src/cluster"
-	cfg "github.com/accuknox/auto-policy-discovery/src/config"
-	"github.com/accuknox/auto-policy-discovery/src/types"
 	"sigs.k8s.io/yaml"
 )
 
+const (
+	RecommendedPolicyTagsAnnotation        = "recommended-policies.accuknox.com/tags"
+	RecommendedPolicyTitleAnnotation       = "policies.kyverno.io/title"
+	RecommendedPolicyDescriptionAnnotation = "policies.kyverno.io/description"
+)
+
 var policyRules []types.MatchSpec
 
 func updateRulesYAML(yamlFile []byte) string {
 	policyRules = []types.MatchSpec{}
 	policyRulesJSON, err := yaml.YAMLToJSON(yamlFile)
 	if err != nil {
-		log.Error().Msgf("failed to convert policy rules yaml to json")
+		log.Error().Msgf("failed to convert policy rules yaml to json: %v", err)
 	}
 	var jsonRaw map[string]json.RawMessage
 	err = json.Unmarshal(policyRulesJSON, &jsonRaw)
 	if err != nil {
-		log.Error().Msgf("failed to unmarshal policy rules json")
+		log.Error().Msgf("failed to unmarshal policy rules json: %v", err)
```

Comment:
return would be needed, since otherwise it would result in a segfault on subsequent loc.
```suggestion
		log.Error().Msgf("failed to unmarshal policy rules json: %v", err)
		return ""
```

Reply:
Yup, error handling was not proper in this function, changed it.